src/devices/oregon_scientific.c

```diff
@@ -871,13 +871,13 @@ static int oregon_scientific_v3_decode(r_device *decoder, bitbuffer_t *bitbuffer
         int id       = msg[2] << 8 | (msg[1] & 0xF0);
         int batt_low = (msg[3] & 0x40)?1:0; // 8th bit instead of 6th commonly used for other devices
 
-        unsigned short int ipower1 = cm180i_power(msg,0);
-        unsigned short int ipower2 = cm180i_power(msg,1);
-        unsigned short int ipower3 = cm180i_power(msg,2);
-        unsigned long long itotal= 0;
+        unsigned ipower1 = cm180i_power(msg,0);
+        unsigned ipower2 = cm180i_power(msg,1);
+        unsigned ipower3 = cm180i_power(msg,2);
+        uint64_t itotal= 0;
         if (msg_len >= 140) itotal= cm180i_total(msg);
 
-        // per hour and in kilowat
+        // per hour and in kilowatt (hence kwh)
```

Comment:
kilowatt, even spelled correctly, is not a unit of energy.  I am not sure what's going on here, but it seems at least not quite right.

Reply:
I did not want to change the original text too much, fix at least the obvious typo and remind that the result is kWh according to the data label further below.

Fixing this fishy thing is not the goal of the PR.  cm180i_total probably returns  Ws (so Joules), with 3600Ws = 1Wh, and to get kWh that's divided by 1000.

The comment could be:

```c
// Convert itotal which is in Ws (or J) to kWh unit.
```

Reply:
There was a [lengthy discussion](https://github.com/merbanan/rtl_433/pull/2418#discussion_r1133254997) about the CM160, IIRC the calculation works by chance only.

Reply:
The new comment looks fine.   I am ok with 1) don't touch it or 2) leave it in a good state, and don't like the middle ground...

Reply:
I updated the commit where is made the first change.